### DISCUSSION TITLE
Livewire\Exceptions\CorruptComponentPayloadException Livewire encountered corrupt data when trying to hydrate the [test] component. Ensure that the [name, id, data] of the Livewire component wasn't tampered with between requests.

### DISCUSSION THREAD

**Discussion by srk82mwi** [livewire/livewire#5528] [1. Help]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <title>Test</title>
    @livewireStyles
</head>
<body>
    <input wire:key="" type="text" wire:model="txtMsg">
    {{$txtMsg}}
    @livewireScripts
</body>
</html>

<?php

namespace App\Http\Livewire;
use Livewire\Component;

class Test extends Component {

    public $txtMsg;

    public function render() { 
        return view('livewire.test');
    }

}
![livewire-error1](https://user-images.githubusercontent.com/123530735/214485424-3c086635-01a9-489b-b780-933a89a4b248.PNG)
![output1](https://user-images.githubusercontent.com/123530735/214485527-b336a0c2-1ad7-4673-b828-19edc6f0badd.PNG)

_Originally posted by @srk82mwi in https://github.com/livewire/livewire/issues/354#issuecomment-1403108258_

**Comment by joshhanley** [comment]
This issue has been converted to a discussion as we are now using discussions to track bugs and ideas.
See this announcement for more details https://github.com/livewire/livewire/discussions/3147

**Comment by joshhanley** [comment]
@srk82mwi it looks like there might be an issue with the code in your description. When you paste code into github, you need to surround it in triple back ticks and then specify the file type like php or blade. That way it gets formatted correctly.

For example
\```php
\\ PHP code here
\```

\```blade
\\ HTML and blade code here
\```

Can you edit your description and fix it up so we can help you? Thanks!

**Comment by winterant** [comment]
I have the same problem. And I found a solution ( [at stackoverflow](https://stackoverflow.com/questions/63848826/livewire-encountered-corrupt-data-when-trying-to-hydrate-the-component) ), but it doesn't work for me....

Any bro has a solution?

**Comment by hugronaphor** [comment]
In majority of cases it's related to adding values to public properties with keys (e.g float) which are not supported well by JS.

**Comment by mcpuishor** [comment]
I had a situation happening for an annoyingly long time, until I realized that I was passing properties to the component that were not defined in the component. 

Livewire would silently ignore this fact on first render, however when the component would update, the not-declared-but-passed-on property would not be there anymore when the component hydrates. 

The component would fail the checksum match and throw the infamous error.
